src/freenet/client/async/ClientPutter.java

```diff
@@ -287,7 +287,7 @@ public void onSuccess(ClientPutState state, ClientContext context) {
 		}
 		if(super.failedBlocks > 0 || super.fatallyFailedBlocks > 0 || super.successfulBlocks < super.totalBlocks) {
 			// USK auxiliary inserts are allowed to fail.
-			if(!uri.isUSK())
+			if((!uri.isUSK()) && !ctx.getCHKOnly)
```

Comment:
I don't understand - why is it okay for a CHK-only insert to have failed like this?